caddyconfig/httpcaddyfile/directives.go

```diff
@@ -265,6 +265,13 @@ func (h Helper) NewBindAddresses(addrs []string) []ConfigValue {
 	return []ConfigValue{{Class: "bind", Value: addrs}}
 }
 
+// WithDispenser returns a new instance based on d. All others Helper
+// fields are copied, so typically maps are shared with this new instance.
+func (h Helper) WithDispenser(d *caddyfile.Dispenser) Helper {
+	h.Dispenser = d
+	return h
+}
```

Comment:
I'm not entirely clear on why this is necessary?

Reply:
The problem is `httpcaddyfile.ParseSegmentAsSubroute` takes a `Helper` rather than a `Dispenser`, which makes it necessary to chain parsing for other directives.

Reply:
Thanks again @francislavoie, yes this simple method is to simplify code in such cases.